contributors/design-proposals/kubelet-authorizer.md

```diff
@@ -0,0 +1,119 @@
+# Scoped Kubelet API Access
+
+Author: Jordan Liggitt (jliggitt@redhat.com)
+
+## Overview
+
+Kubelets are responsible for:
+* creating and updating status of their Node API object
+* updating status of Pod API objects bound to their node
+* creating/deleting "mirror pod" API objects for statically-defined pods running on their node
+* reading Secret and ConfigMap objects referenced by pod specs bound to their node
+
+Currently, kubelets have read/write access to all Node and Pod objects, and read access to all Secret and ConfigMap objects.
+This means that compromising a node gives access to credentials with power to make escalating API calls that affect other nodes.
+
+This proposal limits kubelets' API access using a new node authorizer and admission plugin:
+* Node authorizer
+  * Authorizes requests from nodes using the existing policy rules developed for the `system:node` cluster role
+  * Further restricts secret and configmap requests to only authorize reading objects referenced by pods bound to the node
+* Node admission
+  * Limits a node to mutating its own Node API object
+  * Limits a node to mutating pods bound to itself
+
+## Node identification
+
+The first step is to identify whether a particular API request is being made by 
+a node, and if so, from which node.
+
+The proposed node authorizer and admission plugin will take a `NodeIdentifier` interface:
+
+```go
+type NodeIdentifier interface {
+  // IdentifyNode determines node information from the given user.Info.
+  // nodeName is the name of the Node API object associated with the user.Info,
+  // and may be empty if a specific node cannot be determined.
+  // isNode is true if the user.Info represents an identity issued to a node.
+  IdentifyNode(user.Info) (nodeName string, isNode bool)
+}
+```
+
+The default `NodeIdentifier` implementation:
+* `isNode` - true if the user groups contain the `system:nodes` group
+* `nodeName` - populated if `isNode` is true, and the user name is in the format `system:node:<nodeName>`
```

Comment:
no, the nodeName is used to filter pods, and the kubelet already knows it (since it is responsible for updating the Node status and querying bound pods).